pkg/cmd/run.go

```diff
@@ -107,6 +108,7 @@ func RunMicroshift(cfg *config.MicroshiftConfig, flags *pflag.FlagSet) error {
 	util.Must(m.AddService((controllers.NewVersionManager((cfg)))))
 	util.Must(m.AddService(kustomize.NewKustomizer(cfg)))
 	util.Must(m.AddService(node.NewKubeletServer(cfg)))
+	util.Must(m.AddService(loadbalancerservice.NewLoadbalancerServiceController(cfg)))
```

Comment:
Should the loadbalancer service start before CNI (infrastructureService)?

Reply:
Why/where does it depend on OVN-k?

I believe we should make an agnostic implementation. In the end this would just work by having an nft table / iptable rediecting from the LB port into the NodePort for the host network, no matter what CNI we have.

Otherwise it will complicate the history of CNI pluggability.

Reply:
@pliurh btw, very cool to see work on this :-) we have been missing it for a long time!

Reply:
OVN-K will add the iptable rules. It will add a DNAT rule to redirect the LB port to the clusterIP.

I don't think a complete CNI agnostic way is realistic. We cannot assume what a CNI will do with the LB service. An agnostic way may conflict with the per-CNI behavior.

Reply:
Yes, that's partly true. May be for other CNIs we will have to add something that does the same handling in terms of iptables.